docs/DeveloperGuide.md

```diff
@@ -2,28 +2,131 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Help menu feature
+
+#### Implementation
+
+The "help" feature is facilitated by the  `Help` class.
+It provides a static method `printHelp` to print out a guide on how to use the commands in the application.
+`printHelp` can be used in the event the user issues an invalid command
+
+### Group Creation feature
+
+#### Implementation
+
+The "Group Creation" feature is facilitated by the `Group` class. It provides methods to create a new group and manage
+group membership. The implementation of this feature is as follows:
+
+The Group class maintains a list of members as a `private List<User>` field called `members`.
+
+The `createGroup(String groupName)` method is responsible for creating a new group. It performs the following steps:
+
+1. Checks if a group with the given `groupName` already exists using the `isGroup(String groupName)` method.
+2. If the group does not exist, creates a new `Group` object with the provided `groupName`.
+3. Prints a success message indicating that the group has been created.
+4. Adds the new `Group` object to the `groups` list.
+5. Returns the newly created `Group` object.
+6. If the group already exists, prints an error message indicating that the group already exists.
+
+### Add Member to Group feature
```

Comment:
it is not very clear how the user commands are parsed, as there is no information about how the basic components (such as Ui, Model, Logic, Storage) interact with each other before explaining the features